packages/wrangler/src/pages.tsx

```diff
@@ -1198,15 +1198,33 @@ const createDeployment: CommandModule<
         },
         base64: true,
       }));
+
+      let attempts = 0;
+      const doUpload = async (): Promise<void> => {
+        try {
+          return await fetchResult(`/pages/assets/upload`, {
```

Comment:
If this returns with `success: false`, does that throw?

Reply:
No. Weirdly. I tried `new Response` with status 500, all kinds of things. The only way I could get this to actually loop was an exception within the test :/

Reply:
Oh well

Reply:
`setMockResponse()` always returns a response with `success: true`.
If you want to simulate returning something else then you should consider `setMockRawResponse()`.

Reply:
But rest assured, `fetchResponse()` will thrown if `success: false`.

Reply:
Aha now `setMockRawResponse` makes sense to me! Fixed now.